packages/shared/src/components/post/PostCodeSnippets.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { ReactElement, useMemo, useState } from 'react';
+import classNames from 'classnames';
+import { usePostCodeSnippetsQuery } from '../../hooks/post/usePostCodeSnippets';
+import { RenderMarkdown } from '../RenderMarkdown';
+import { Button, ButtonSize, ButtonVariant } from '../buttons/Button';
+import { ArrowIcon } from '../icons';
+
+export type PostCodeSnippetsProps = {
+  className?: string;
+  postId: string | undefined;
+};
+
+export const PostCodeSnippets = ({
+  className,
+  postId,
+}: PostCodeSnippetsProps): ReactElement => {
+  const { data, fetchNextPage } = usePostCodeSnippetsQuery({
+    postId,
+  });
+
+  const [activeSnippetIndex, setActiveSnippetIndex] = useState<number>(0);
+
+  const pagesFlat = useMemo(() => {
+    return data?.pages.flatMap((page) => page.edges);
+  }, [data]);
+
+  const codeSnippet = useMemo(() => {
+    return pagesFlat?.[activeSnippetIndex]?.node;
+  }, [activeSnippetIndex, pagesFlat]);
+
+  if (!codeSnippet) {
+    return null;
+  }
+
+  const markdownContent = `\`\`\`\n${codeSnippet.content}\n\`\`\``;
```

Comment:
Don't know if our markdown renderer supports it, but instead of escaping the backticks, we could use tildes `~~~`

This is valid
```
~~~typescript
const tilde: string
~~~
```